exporter/otlphttpexporter/otlp.go

```diff
@@ -301,21 +301,20 @@ func handlePartialSuccessResponse(resp *http.Response, partialSuccessHandler par
 type partialSuccessHandler func(bytes []byte, contentType string) error
 
 func tracesPartialSuccessHandler(responseBytes []byte, contentType string) error {
-	if contentType != protobufContentType && contentType != jsonContentType {
-		return nil
-	}
 	exportResponse := ptraceotlp.NewExportResponse()
-
-	if contentType == protobufContentType {
+	switch contentType {
+	case protobufContentType:
 		err := exportResponse.UnmarshalProto(responseBytes)
 		if err != nil {
 			return fmt.Errorf("error parsing protobuf response: %w", err)
 		}
-	} else if contentType == jsonContentType {
+	case jsonContentType:
 		err := exportResponse.UnmarshalJSON(responseBytes)
 		if err != nil {
 			return fmt.Errorf("error parsing json response: %w", err)
 		}
+	default:
+		return nil
```

Comment:
I agree that returning nil was the previous default behavior, but I think that was incorrect.  I am ok taking care of that in a separate PR tho, as long as we create an issue to track it.

Reply:
I think this should return an error with unsupported content type now?

Reply:
I'm definitely ok with updating this to return an error as part of this PR.